src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -0,0 +1,258 @@
+import com.cleanroommc.modularui.api.math.Alignment;
+import com.cleanroommc.modularui.api.screen.ModularWindow;
+import com.cleanroommc.modularui.api.screen.UIBuildContext;
+import com.cleanroommc.modularui.common.widget.*;
+import com.cleanroommc.modularui.common.widget.textfield.TextFieldWidget;
+import gregtech.api.util.GTLog;
+import gregtech.client.renderer.texture.Textures;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.*;
+
+
+public class CoverDetectorEnergyAdvanced extends CoverBehavior implements CoverWithUI, ITickable{
+    public int minPercent, maxPercent;
+    public int minEU, maxEU;
+    private int outputAmount;
+    private boolean inverted;
+    private boolean useEU;
+    private int maxLength,  maxEnterable;
+
+    public CoverDetectorEnergyAdvanced (ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        this.minPercent = 10;
+        this.maxPercent = 90;
+        this.minEU = 512;
+        this.maxEU = 2048;
+        this.outputAmount = 0;
+        this.inverted = false;
+        this.useEU = false;
+        this.maxLength = 3;
+        this.maxEnterable = 100;
+    }
+
+    @Override
+    public boolean canAttach() {
+        return coverHolder.getCapability(GregtechCapabilities.CAPABILITY_ENERGY_CONTAINER, null) != null;
+    }
+
+    // TODO: use unique texture for advanced energy detector
+    @Override
+    public void renderCover(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline, Cuboid6 plateBox, BlockRenderLayer layer) {
+        Textures.DETECTOR_ENERGY.renderSided(attachedSide, plateBox, renderState, pipeline, translation);
+    }
+
+    @Override
+    public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, CuboidRayTraceResult hitResult){
+        if (!this.coverHolder.getWorld().isRemote) {
+            GregTechUI.getCoverUi(attachedSide).open(playerIn, coverHolder.getWorld(), coverHolder.getPos());
+        }
+        return EnumActionResult.SUCCESS;
+    }
+
+    @Override
+    public void update() {
+        if (coverHolder.getOffsetTimer() % 20 != 0)
+            return;
+
+        IEnergyContainer energyContainer = coverHolder.getCapability(GregtechCapabilities.CAPABILITY_ENERGY_CONTAINER, null);
+        if (energyContainer != null && energyContainer.getEnergyCapacity() > 0){
+            float currentStorageRatio = 100f * energyContainer.getEnergyStored() / energyContainer.getEnergyCapacity();
+
+            if (useEU) {
+                compareValue((float) energyContainer.getEnergyStored(), maxEU, minEU);
+            } else {
+                compareValue(currentStorageRatio, maxPercent, minPercent);
+            }
+            setRedstoneSignalOutput(outputAmount);
+        }
+    }
+
+    private void compareValue (float value, int maxValue, int minValue) {
```

Comment:
Would it be useful to emit differing redstone levels dependent upon the amount of difference between the read value and the max/min value of the cover?

Reply:
I'm not sure what you mean by this, can you elaborate?

Reply:
No, I meant something like: EU Range is 0 to 2048, at 1024 EU, emit a signal strength of 8, at 512, a signal strength of 4, etc.

Basically scaling the output strength of the signal by how close the value read from the machine is to the defined minimum and maximum.

Reply:
Ah, I see what you mean now. 

I think I've implemented this feature now, behind a toggle button in case it's undesirable for the player.